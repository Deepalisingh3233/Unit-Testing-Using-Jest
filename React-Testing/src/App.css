*{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.btn{
  margin-top: 10px;
}

.main{
  margin: 30px;
}

h1{
  margin-bottom: 10px;
}
h2{
  margin: 10px;
}
