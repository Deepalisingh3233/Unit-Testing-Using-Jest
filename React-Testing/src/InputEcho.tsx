import React, {useState} from "react";


export default function InputEcho() {
  const [inputValue, setInputValue] = useState('')
  const [displayValue, setDisplayValue] = useState('')

  const handleClick = () => {
    setDisplayValue(inputValue)
  }

  return (
    <>
    <div>
      <input
        type="text"
        placeholder="Type something..."
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />
      <button onClick={handleClick}>Show Text</button>
      <p data-testid="output">{displayValue}</p>
    </div>
    </>
  )
}

