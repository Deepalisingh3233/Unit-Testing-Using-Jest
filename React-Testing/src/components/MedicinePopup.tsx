import React, { useState } from 'react';
import { X } from 'lucide-react';

const medicines = [
  {
    storedAt: 'S12/Rac4',
    batchNo: 'AXB10851',
    mfgDate: 'JAN-2024',
    expDate: 'JAN-2026',
    mrp: 237.0,
    unitPerQty: 10,
    avlQty: 10,
  },
  {
    storedAt: 'S6/Rac6',
    batchNo: 'EXB10605',
    mfgDate: 'JAN-2024',
    expDate: 'JAN-2026',
    mrp: 201.0,
    unitPerQty: 10,
    avlQty: 150,
  },
  {
    storedAt: 'S6/Rac5',
    batchNo: 'BXB10882',
    mfgDate: 'JAN-2024',
    expDate: 'JAN-2026',
    mrp: 198.2,
    unitPerQty: 10,
    avlQty: 210,
  },
];

export default function MedicinePopup({ onClose, onAdd }) {
  const [quantities, setQuantities] = useState(medicines.map(() => 0));

  const handleQtyChange = (index, value) => {
    const newQuantities = [...quantities];
    newQuantities[index] = parseInt(value) || 0;
    setQuantities(newQuantities);
  };

  const handleAdd = () => {
    const selected = medicines
      .map((med, index) => ({ ...med, qty: quantities[index] }))
      .filter(med => med.qty > 0);
    onAdd(selected);
  };

  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black bg-opacity-40 z-50">
      <div className="bg-white p-6 rounded-2xl shadow-xl w-full max-w-4xl relative overflow-x-auto">
        <button
          className="absolute top-4 right-4 text-gray-600 hover:text-black"
          onClick={onClose}
        >
          <X />
        </button>
        <h2 className="text-2xl font-bold mb-1">Augmentin 625 Duo Tablet</h2>
        <p className="mb-4 text-gray-600">Amoxycillin (500mg) + Clavulanic Acid (125mg)</p>

        <table className="w-full border-collapse text-sm">
          <thead className="bg-teal-600 text-white">
            <tr>
              <th className="p-2 text-left">Stored At</th>
              <th className="p-2 text-left">Batch No.</th>
              <th className="p-2 text-left">Mfg. Date</th>
              <th className="p-2 text-left">Exp. Date</th>
              <th className="p-2 text-left">MRP (â‚¹)</th>
              <th className="p-2 text-left">Unit Per Qty.</th>
              <th className="p-2 text-left">Avl. Qty.</th>
              <th className="p-2 text-left">Qty.</th>
            </tr>
          </thead>
          <tbody>
            {medicines.map((med, index) => (
              <tr key={index} className="border-b">
                <td className="p-2">{med.storedAt}</td>
                <td className="p-2">{med.batchNo}</td>
                <td className="p-2">{med.mfgDate}</td>
                <td className="p-2">{med.expDate}</td>
                <td className="p-2">{med.mrp.toFixed(1)}</td>
                <td className="p-2">{med.unitPerQty}</td>
                <td className="p-2">{med.avlQty}</td>
                <td className="p-2">
                  <input
                    type="number"
                    min="0"
                    max={med.avlQty}
                    className="w-16 border rounded px-2 py-1"
                    value={quantities[index]}
                    onChange={(e) => handleQtyChange(index, e.target.value)}
                  />
                </td>
              </tr>
            ))}
          </tbody>
        </table>

        <div className="mt-4 flex justify-end">
          <button
            className="bg-teal-600 text-white px-6 py-2 rounded hover:bg-teal-700"
            onClick={handleAdd}
          >
            +Add
          </button>
        </div>
      </div>
    </div>
  );
}
