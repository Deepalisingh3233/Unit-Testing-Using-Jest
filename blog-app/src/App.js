import logo from './logo.svg';
import './App.css';
import { useState } from 'react';

function App() {
  const [data, setData] = useState("");
  return (
    <div className="App">
      <h1>Test OnChange Event with Input Text</h1>
      <input type='text' value={data} onChange={(e) => setData(e.target.value+"test")}/>

      <h1>Test Click Event with Button</h1>
      <button onClick={() => setData("updated data")}>Update Data</button>
      <h1>{data}</h1>
    </div>
  );
}

// function App() {
//   return (
//     <div className="App">
//       <p>First React Test CASE</p>
//       <p>Deepali Singh</p>
//       <input type='text' 
//       readOnly 
//       placeholder='Enter user name' 
//       name='username' 
//       id='userID' 
//       value='Deepali Singh' />
//       <br/>
//       <br/>
//       <br/>
      
//       <img
//         src="https://images.pexels.com/photos/17690714/pexels-photo-17690714.jpeg"
//         title="Beautiful girl in the park"
//         width="500"
//       />
//     </div>
//   );
// }

export default App;
