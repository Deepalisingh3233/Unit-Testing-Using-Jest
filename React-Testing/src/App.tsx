import './App.css';
import MedicinePopup from './components/MedicinePopup';
import InputEcho from './InputEcho';

const user = {
  name: 'Ankita Bajpai'
};

function handleClick() {
  alert("Hello, I am " + user.name);
}

function MyButton() {
  return (
    <button className="btn" onClick={handleClick}>
      Click Me!
    </button>
  );
}

export default function App() {

  return (
    <>
    <div className="main">
      <h1>{user.name}</h1>
      <img
        className="avatar"
        src="https://i.imgur.com/yXOvdOSs.jpg"
        alt={`Photo of ${user.name}`}
        style={{
          width: 90,
          height: 90,
          borderRadius: '50%',
        }}
      />
      <div>
        <MyButton />
      </div>

      <div>
        <h2>Type Something:</h2>
        <InputEcho/> 
      </div>

      <div><MedicinePopup/></div>
    </div>
    </>
  );
}
