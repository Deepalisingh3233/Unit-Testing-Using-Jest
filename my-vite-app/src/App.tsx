import React, { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import { Counter } from './components/Counter'
import InputEcho from './components/InputEcho'
import MedicinePopup from './components/MedicinePopup'

function App() {
  const [count, setCount] = useState(0)

  return (
    <>
     <Counter />
     <br/>
     <InputEcho />
     <br/>
     <MedicinePopup/>
    </>
  )
}

export default App
