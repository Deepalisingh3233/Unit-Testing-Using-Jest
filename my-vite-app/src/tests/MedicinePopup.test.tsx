import { fireEvent, render, screen, within } from "@testing-library/react";
import App from "../App";
import React from "react";
import { Target } from "lucide-react";
import InputEcho from "../components/InputEcho";
import MedicinePopup from "../components/MedicinePopup";



describe("Validate the table headings", () => {
    test("Check Medicine name is present or not", () => {
        render(<App />)
        const medicineName = screen.getByText("Augmentin 625 Duo Tablet");
        expect(medicineName).toBeInTheDocument();
    })
    test("Check Medicine salt is present or not", () => {
        render(<App />)
        const medicineSalt = screen.getByText("Amoxycillin (500mg) + Clavulanic Acid (125mg)");
        expect(medicineSalt).toBeInTheDocument();
    })
    test("Check Stored At heading is present or not", () => {
        render(<App />)
        const storedAtHeading = screen.getByText("Stored At");
        expect(storedAtHeading).toBeInTheDocument();
    })
    test("Check Batch No. heading is present or not", () => {
        render(<App />)
        const batchNoHeading = screen.getByText("Batch No.");
        expect(batchNoHeading).toBeInTheDocument();
    })
    test("Check Mfg. Date heading is present or not", () => {
        render(<App />)
        const mfgDateHeading = screen.getByText("Mfg. Date");
        expect(mfgDateHeading).toBeInTheDocument();
    })
    test("Check Exp. Date heading is present or not", () => {
        render(<App />)
        const expDateHeading = screen.getByText("Exp. Date");
        expect(expDateHeading).toBeInTheDocument();
    })
    test("Check MRP heading is present or not", () => {
        render(<App />)
        const mrpHeading = screen.getByText("MRP (â‚¹)");
        expect(mrpHeading).toBeInTheDocument();
    })
    test("Check Unit per Qty. heading is present or not", () => {
        render(<App />)
        const unitPerQtyHeading = screen.getByText("Unit Per Qty.");
        expect(unitPerQtyHeading).toBeInTheDocument();
    })
    test("Check Avl. Qty. heading is present or not", () => {
        render(<App />)
        const avlQtyHeading = screen.getByText("Avl. Qty.");
        expect(avlQtyHeading).toBeInTheDocument();
    })
    test("Check Qty. heading is present or not", () => {
        render(<App />)
        const qtyHeading = screen.getByText("Qty.");
        expect(qtyHeading).toBeInTheDocument();
    })
});

describe("Validate the Table data", () => {
    test("Check table body contains data", () => {
        render(<App />);
        const table = screen.getByRole("table");
        const tbody = within(table).getByTestId("table-body");
        const tdElements = within(tbody).getAllByRole("cell");
        expect(tdElements.length).toBeGreaterThan(0);
    })
    test("Check the Qty. input field present or not in table", () => {
        render(<App />);
        const table = screen.getByRole("table");
        const qtyInputbox = within(table).getAllByRole("spinbutton");
        expect(qtyInputbox.length).toBeGreaterThan(0);
    })
    test("Check the functionality of Qty. input box", () => {
        render(<App />)
        const table = screen.getByRole("table");
        const qtyInputbox = screen.getAllByRole("spinbutton");
        fireEvent.change(qtyInputbox[1], {target: {value: "123"}});
    })
})

describe("Validate the Add button", () => {
     test("Check that Add button is present or not", () => {
        render(<App />)
        const addBtn = screen.getByRole("button", {name: "+Add", hidden: true});
        expect(addBtn).toBeInTheDocument();
    })
    /* test("Check that the Add button is clickable or not", () => {
        render(<App />);
         const mockOnAdd = vi.fn(); // or jest.fn() if using Jest
        const mockOnClose = vi.fn();

        render(<App onAdd={mockOnAdd} onClose={mockOnClose} />);
        const addBtn = screen.getAllByRole("button", {name: "+Add", hidden: true});
        // expect(addBtn).toBeEnabled();
        fireEvent.click(addBtn[1]);
        expect(mockOnAdd).toHaveBeenCalled();
    }) */
})

test("Snapshot for the app component", () => {
    const container = render(<App />);
    expect(container).toMatchSnapshot();
})