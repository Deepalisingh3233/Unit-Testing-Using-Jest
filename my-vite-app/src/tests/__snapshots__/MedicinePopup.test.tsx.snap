// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot for the app component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          data-testid="increment"
        >
          Count: 
          0
        </button>
      </div>
      <br />
      <div>
        <input
          placeholder="Type something..."
          type="text"
          value=""
        />
        <button>
          Show Text
        </button>
        <p
          data-testid="output"
        />
      </div>
      <br />
      <div
        class="fixed inset-0 flex bg-black bg-opacity-40 z-50"
      >
        <div
          class="bg-white p-6 rounded-2xl shadow-xl w-full max-w-4xl relative overflow-x-auto"
        >
          <h2
            class="text-2xl font-bold mb-1"
          >
            Augmentin 625 Duo Tablet
          </h2>
          <p
            class="mb-4 text-gray-600"
          >
            Amoxycillin (500mg) + Clavulanic Acid (125mg)
          </p>
          <table
            class="w-full border-collapse text-sm"
          >
            <thead
              class="bg-teal-600 text-white"
            >
              <tr>
                <th
                  class="p-2 text-left"
                >
                  Stored At
                </th>
                <th
                  class="p-2 text-left"
                >
                  Batch No.
                </th>
                <th
                  class="p-2 text-left"
                >
                  Mfg. Date
                </th>
                <th
                  class="p-2 text-left"
                >
                  Exp. Date
                </th>
                <th
                  class="p-2 text-left"
                >
                  MRP (₹)
                </th>
                <th
                  class="p-2 text-left"
                >
                  Unit Per Qty.
                </th>
                <th
                  class="p-2 text-left"
                >
                  Avl. Qty.
                </th>
                <th
                  class="p-2 text-left"
                >
                  Qty.
                </th>
              </tr>
            </thead>
            <tbody
              data-testid="table-body"
            >
              <tr
                class="border-b"
              >
                <td
                  class="p-2"
                >
                  S12/Rac4
                </td>
                <td
                  class="p-2"
                >
                  AXB10851
                </td>
                <td
                  class="p-2"
                >
                  JAN-2024
                </td>
                <td
                  class="p-2"
                >
                  JAN-2026
                </td>
                <td
                  class="p-2"
                >
                  237.0
                </td>
                <td
                  class="p-2"
                >
                  10
                </td>
                <td
                  class="p-2"
                >
                  10
                </td>
                <td
                  class="p-2"
                >
                  <input
                    class="w-16 border rounded px-2 py-1"
                    max="10"
                    min="0"
                    type="number"
                    value="0"
                  />
                </td>
              </tr>
              <tr
                class="border-b"
              >
                <td
                  class="p-2"
                >
                  S6/Rac6
                </td>
                <td
                  class="p-2"
                >
                  EXB10605
                </td>
                <td
                  class="p-2"
                >
                  JAN-2024
                </td>
                <td
                  class="p-2"
                >
                  JAN-2026
                </td>
                <td
                  class="p-2"
                >
                  201.0
                </td>
                <td
                  class="p-2"
                >
                  10
                </td>
                <td
                  class="p-2"
                >
                  150
                </td>
                <td
                  class="p-2"
                >
                  <input
                    class="w-16 border rounded px-2 py-1"
                    max="150"
                    min="0"
                    type="number"
                    value="0"
                  />
                </td>
              </tr>
              <tr
                class="border-b"
              >
                <td
                  class="p-2"
                >
                  S6/Rac5
                </td>
                <td
                  class="p-2"
                >
                  BXB10882
                </td>
                <td
                  class="p-2"
                >
                  JAN-2024
                </td>
                <td
                  class="p-2"
                >
                  JAN-2026
                </td>
                <td
                  class="p-2"
                >
                  198.2
                </td>
                <td
                  class="p-2"
                >
                  10
                </td>
                <td
                  class="p-2"
                >
                  210
                </td>
                <td
                  class="p-2"
                >
                  <input
                    class="w-16 border rounded px-2 py-1"
                    max="210"
                    min="0"
                    type="number"
                    value="0"
                  />
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="mt-4 flex justify-end"
          >
            <button
              class="bg-teal-600 text-white px-6 py-2 rounded hover:bg-teal-700"
            >
              +Add
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        data-testid="increment"
      >
        Count: 
        0
      </button>
    </div>
    <br />
    <div>
      <input
        placeholder="Type something..."
        type="text"
        value=""
      />
      <button>
        Show Text
      </button>
      <p
        data-testid="output"
      />
    </div>
    <br />
    <div
      class="fixed inset-0 flex bg-black bg-opacity-40 z-50"
    >
      <div
        class="bg-white p-6 rounded-2xl shadow-xl w-full max-w-4xl relative overflow-x-auto"
      >
        <h2
          class="text-2xl font-bold mb-1"
        >
          Augmentin 625 Duo Tablet
        </h2>
        <p
          class="mb-4 text-gray-600"
        >
          Amoxycillin (500mg) + Clavulanic Acid (125mg)
        </p>
        <table
          class="w-full border-collapse text-sm"
        >
          <thead
            class="bg-teal-600 text-white"
          >
            <tr>
              <th
                class="p-2 text-left"
              >
                Stored At
              </th>
              <th
                class="p-2 text-left"
              >
                Batch No.
              </th>
              <th
                class="p-2 text-left"
              >
                Mfg. Date
              </th>
              <th
                class="p-2 text-left"
              >
                Exp. Date
              </th>
              <th
                class="p-2 text-left"
              >
                MRP (₹)
              </th>
              <th
                class="p-2 text-left"
              >
                Unit Per Qty.
              </th>
              <th
                class="p-2 text-left"
              >
                Avl. Qty.
              </th>
              <th
                class="p-2 text-left"
              >
                Qty.
              </th>
            </tr>
          </thead>
          <tbody
            data-testid="table-body"
          >
            <tr
              class="border-b"
            >
              <td
                class="p-2"
              >
                S12/Rac4
              </td>
              <td
                class="p-2"
              >
                AXB10851
              </td>
              <td
                class="p-2"
              >
                JAN-2024
              </td>
              <td
                class="p-2"
              >
                JAN-2026
              </td>
              <td
                class="p-2"
              >
                237.0
              </td>
              <td
                class="p-2"
              >
                10
              </td>
              <td
                class="p-2"
              >
                10
              </td>
              <td
                class="p-2"
              >
                <input
                  class="w-16 border rounded px-2 py-1"
                  max="10"
                  min="0"
                  type="number"
                  value="0"
                />
              </td>
            </tr>
            <tr
              class="border-b"
            >
              <td
                class="p-2"
              >
                S6/Rac6
              </td>
              <td
                class="p-2"
              >
                EXB10605
              </td>
              <td
                class="p-2"
              >
                JAN-2024
              </td>
              <td
                class="p-2"
              >
                JAN-2026
              </td>
              <td
                class="p-2"
              >
                201.0
              </td>
              <td
                class="p-2"
              >
                10
              </td>
              <td
                class="p-2"
              >
                150
              </td>
              <td
                class="p-2"
              >
                <input
                  class="w-16 border rounded px-2 py-1"
                  max="150"
                  min="0"
                  type="number"
                  value="0"
                />
              </td>
            </tr>
            <tr
              class="border-b"
            >
              <td
                class="p-2"
              >
                S6/Rac5
              </td>
              <td
                class="p-2"
              >
                BXB10882
              </td>
              <td
                class="p-2"
              >
                JAN-2024
              </td>
              <td
                class="p-2"
              >
                JAN-2026
              </td>
              <td
                class="p-2"
              >
                198.2
              </td>
              <td
                class="p-2"
              >
                10
              </td>
              <td
                class="p-2"
              >
                210
              </td>
              <td
                class="p-2"
              >
                <input
                  class="w-16 border rounded px-2 py-1"
                  max="210"
                  min="0"
                  type="number"
                  value="0"
                />
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="mt-4 flex justify-end"
        >
          <button
            class="bg-teal-600 text-white px-6 py-2 rounded hover:bg-teal-700"
          >
            +Add
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
